AppCUI/src/Controls/Grid.cpp

```diff
@@ -0,0 +1,1162 @@
+                    break;
+                default:
+                    break;
+                }
+            }
+            else if (topRight == context->anchorCellIndex)
+            {
+                switch (keyCode)
+                {
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Left:
+                    if (xLeft > 0)
+                        xLeft -= 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Right:
+                    if (xLeft < context->columnsNo - 1)
+                        xLeft += 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Up:
+                    if (yBot > 0)
+                        yBot -= 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Down:
+                    if (yBot < context->rowsNo - 1)
+                        yBot += 1;
+                    break;
+                default:
+                    break;
+                }
+            }
+            else if (bottomLeft == context->anchorCellIndex)
+            {
+                switch (keyCode)
+                {
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Left:
+                    if (xRight > 0)
+                        xRight -= 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Right:
+                    if (xRight < context->columnsNo - 1)
+                        xRight += 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Up:
+                    if (yTop > 0)
+                        yTop -= 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Down:
+                    if (yTop < context->rowsNo - 1)
+                        yTop += 1;
+                    break;
+                default:
+                    break;
+                }
+            }
+            else if (bottomRight == context->anchorCellIndex)
+            {
+                switch (keyCode)
+                {
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Left:
+                    if (xLeft > 0)
+                        xLeft -= 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Right:
+                    if (xLeft < context->columnsNo - 1)
+                        xLeft += 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Up:
+                    if (yTop > 0)
+                        yTop -= 1;
+                    break;
+                case AppCUI::Input::Key::Shift | AppCUI::Input::Key::Down:
+                    if (yTop < context->rowsNo - 1)
+                        yTop += 1;
+                    break;
+                default:
+                    break;
+                }
```

Comment:
am inteles codul --> dar e foarte greu de citit.
Chiar daca nu e urgent --> nu ar merge sa il spargem in niste functii (gen de miscare, de selectie, etc) ?

Reply:
e mai ok sa l sparg acum

Reply:
done